[feature/appshared-reorg] ownCloudAppShared.framework reorganisation

felix-schwarz opened the pull request:
## Description
This PR reorganizes the source code to move key reusable Swift code to the ownCloudAppShared.framework and performs some housekeeping and modernizations along the way.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

felix-schwarz committed:
- remove UIApplication.shared dependency
	- new UserInterfaceContext class, providing access to what UIApplication (and extensions thereof) were previously accessed
	- apps and extensions can implement an UserInterfaceContext extension implementing the UserInterfaceContextProvider protocol to provide access to supported UI elements
	- allowed removal of UIApplication+Extension
- moving major parts from ownCloud to ownCloudAppShared (major refactoring)
	- declare relevant methods and properties open or public
	- adapt all relevant XIBs
- cleanups
	- move "feedback" settings from MoreSettingsSection to VendorServices ("app") and update CONFIGURATION.md accordingly
	- move all intents from ownCloudAppShared to the intents app extension
	- merge AppLockHelper method into AppLockManager as a class-level property (AppLockManager.isPassCodeEnabled)
	- make UIDevice.current.isIpad a property (rather than a method)
	- fix indentations in many areas

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/ios-app?pullRequest=751) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/ios-app?pullRequest=751) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/ios-app?pullRequest=751) it.</sub>

felix-schwarz committed:
- moving Action base class to ownCloudAppShared
- moving FileListTableViewController to ownCloudAppShared, decouple big dependencies like the more view and open item handling so it can remain in the app itself

felix-schwarz committed:
- fix AppIcon-emm icon set
- QueryListTableViewController:
	- factor out multi selection
	- move to ownCloudAppShared
- adapt and move rest of app and framework code where needed

felix-schwarz committed:
- factor out inline message support for ClientQueryViewController
- move ClientQueryViewController to ownCloudAppShared
- move ClientDirectoryPickerViewController to ownCloudAppShared

felix-schwarz committed:
Merge remote-tracking branch 'origin/milestone/1.4' into feature/appshared-reorg

felix-schwarz committed:
- move NamingViewController and CreateFolderAction to ownCloudAppShared
- move TVG resources to ownCloudAppShared

felix-schwarz commented:
All changes of this PR are also contained in #752, so this can be closed.
